### ISSUE TITLE
TransitionLink Length and Routing

### ISSUE THREAD

**Issue by dtmle** [TylerBarnes/gatsby-plugin-transition-link#79]
I have the following TransitionLink, which routes to a page with a similar TransitionLink.
```
<TransitionLink
  to={`/work/${toSafePath(md.title)}`}
  exit={{
    delay: 0.2,
    length: 0.4,
  }}
  entry={{
    delay: 0,
    length: 0.4
  }}
  onClick={() => set(false)}
>
  <Image
    fluid={md.image.childImageSharp.fluid}
    alt={md.alt}
    imgStyle={{
      objectFit: 'cover',
    }}
  />
</TransitionLink>
```

My TransitionLink on the other page:
```
<TransitionLink
  to='/'
  exit={{
    delay: 0.2,
    length: 0.4,
  }}
  entry={{
    delay: 0,
    length: 0.4,
  }}
  onClick={() => set(false)}
>
  <button>X</button>
</TransitionLink>
```

I am running into an issue where if you click the link, then attempt to click another link very quickly, no routing happens. I am assuming this is because of the length property (0.4), which must be complete before routing again.

I have a fade animation which lasts 0.4 seconds, and happens on routing. If I click the link then attempt to click another link quickly, it fires the animation but doesn't route. This leaves me with a blank page due to the fade animation.

I was able to fix the issue by lowering length to 0.2 for both TransitionLink's entry and exit, though I was under the impression that the length should be the length of your animation. Perhaps I am misunderstanding how length should work with animations?

**Comment by TylerBarnes** [comment]
Yes, the reason you can't click another link before the animation is finished is for that exact reason. Sometimes the app will be in an inconsistent state and the animation will be applied to the wrong page if an animation can be started in the middle of another animation by clicking links quickly.

The length is the length of time that the animation for an individual page goes for. The timing can be hard to visualize since it has to account for a lot of scenarios between two potentially overlapping page animations.

A crude description of the timeline of events can be found at https://transitionlink.tylerbarnes.ca/docs/timeline-of-events/

The total time routing is locked up to prevent the bugs you're seeing is based on the entire length of that timeline, however, for the entering page you're actually able to extend an animation as long as you want since there's no way to prevent that. You could set the entry length to 1 second but then program an animation that lasts for 10 seconds. The best thing to do is to make sure your animation length matches up with the length you've entered onto the transition link.
My workflow for doing that is to use the length prop that gets passed through to my trigger functions or to the page props & transition status component.

TLDR; disabling routing during animations is by design and is the desired behaviour, use the length prop to make sure your animations match up to your transition link settings.

**Comment by dtmle** [comment]
Thank you for your reply.
I initially lined up my animation length with the transition link lengths, which did not resolve this problem. Lowering the length of the link while keeping the same animation length resolved the issue, as well as adding a delay into entry. 

I am also not using TransitionStatus, as I ran into other issues, such as my animation firing it's exit when I go back a page.

I will keep experimenting to see what works best, but I am loving the plugin so far. 
Thanks for all your hard work.

**Comment by TylerBarnes** [comment]
Interesting! Potentially the route locking timing is off then. It would be helpful if I could check that out using your repo, otherwise I can try to reproduce on my own.
